src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cache/ShardRequestCacheRca.java

```diff
@@ -0,0 +1,223 @@
+    private static final Logger LOG = LogManager.getLogger(ShardRequestCacheRca.class);
+    private static final long THRESHOLD_TIME_PERIOD_IN_MILLISECOND = TimeUnit.SECONDS.toMillis(300);
+
+    private final Metric shardRequestCacheEvictions;
+    private final Metric shardRequestCacheHits;
+    private final Metric shardRequestCacheSize;
+    private final Metric shardRequestCacheMaxSize;
+    private final int rcaPeriod;
+    private int counter;
+    private boolean exceedsSize;
+    protected Clock clock;
+    private final CacheCollector cacheEvictionCollector;
+    private final CacheCollector cacheHitCollector;
+
+    public <M extends Metric> ShardRequestCacheRca(final int rcaPeriod, final M shardRequestCacheEvictions,
+                                                   final M shardRequestCacheHits, final M shardRequestCacheSize,
+                                                   final M shardRequestCacheMaxSize) {
+        super(5);
+        this.rcaPeriod = rcaPeriod;
+        this.shardRequestCacheEvictions = shardRequestCacheEvictions;
+        this.shardRequestCacheHits = shardRequestCacheHits;
+        this.shardRequestCacheSize = shardRequestCacheSize;
+        this.shardRequestCacheMaxSize = shardRequestCacheMaxSize;
+        this.counter = 0;
+        this.exceedsSize = Boolean.FALSE;
+        this.clock = Clock.systemUTC();
+        this.cacheEvictionCollector = new CacheCollector(SHARD_REQUEST_CACHE_EVICTION,
+                shardRequestCacheEvictions, THRESHOLD_TIME_PERIOD_IN_MILLISECOND);
+        this.cacheHitCollector = new CacheCollector(SHARD_REQUEST_CACHE_HIT,
+                shardRequestCacheHits, THRESHOLD_TIME_PERIOD_IN_MILLISECOND);
+    }
+
+    @VisibleForTesting
+    public void setClock(Clock clock) {
+        this.clock = clock;
+    }
+
+    @Override
+    public ResourceFlowUnit operate() {
+        counter += 1;
+        long currTimestamp = clock.millis();
+
+        cacheEvictionCollector.collect(currTimestamp);
+        cacheHitCollector.collect(currTimestamp);
+        if (counter >= rcaPeriod) {
+            ResourceContext context;
+            HotNodeSummary nodeSummary;
+
+            ClusterDetailsEventProcessor.NodeDetails currentNode = ClusterDetailsEventProcessor.getCurrentNodeDetails();
+            double cacheSize = getTotalSizeInMB(shardRequestCacheSize);
+            double cacheMaxSize = getTotalSizeInMB(shardRequestCacheMaxSize);
+            exceedsSize = cacheMaxSize != 0 && cacheMaxSize != 0 && cacheSize > cacheMaxSize;
+
+            // if eviction and hit counts persists in last 5 minutes and cache size exceeds the max cache size configured,
+            // the cache is considered as unhealthy
+            if (cacheEvictionCollector.isMetricPresentForThresholdTime(currTimestamp)
+                    && cacheHitCollector.isMetricPresentForThresholdTime(currTimestamp)
+                    && exceedsSize) {
```

Comment:
It's not clear why `exceedsSize` is important for thrashing. 
Is this to remove false positives due to TTL based evictions? The doc string mentions that TTL based evictions are not performed because setting is not present and so we ignore it.

Even to remove the TTL eviction false +ves, cacheSize may not *exceed* maxCacheSize because -
1. ES accounting should mostly be accurate, which prevents cacheSize from exceeding cacheMaxSize. 
2. Evictions should have brought down the cacheSize.

Maybe we should check for cacheSize to be within a threshold of cacheMaxSize: assume evictions were due to TTL if `cacheMaxSize - cacheSize > Threshold`